dictionary.go

```diff
@@ -1242,7 +1242,7 @@ func newIndexedPage(typ *indexedType, columnIndex int16, numValues int32, data e
 			copy(tmp, values)
 			values = tmp
 		} else {
-			clear := values[len(values) : len(values)+size]
+			clear := values[len(values):size]
```

Comment:
I'm making guesses about the intention of this code but I believe it's designed to extend the length of the values slice and then zero it out?

should it be?
```
clear := values[:size]
```

Or perhaps the code is attempting to clear only the extended portion of the values slice? So something like:
```
origSz := len(values)
values = values[:size]
clear := values[origSz:size]
```

Reply:
It extends the slice to `size` further below. So it's clearing out pre-existing values beyond its existing length so when it extends the slice, it doesn't inadvertently "discover" old/unintentional values.

Was the above a suggestion to rewrite this line into the three in your code block? (Unclear if it was meant to be a suggestion or just a question.)

Reply:
oh i see. we know the underlying array is long enough b/c of the `cap(values) < size` check above so the reslice works and then we zero out the values. Then below (as you pointed out) is where we reslice values to be the requested size.

i read the change too fast.